### PR TITLE
Update Teku to 23.6.0

### PR DESCRIPTION
**Pull request by jframe**
## PR Description
Update Teku to 23.6.0

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/DataStructureUtilAdapter.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ private ExecutionPayload randomExecutionPayload(final SpecVersion specVersion) {
                     .blockHash(util.randomBytes32())
                     .transactions(util.randomExecutionPayloadTransactions())
                     .withdrawals(() -> randomExecutionPayloadWithdrawals(specVersion))
-                    .excessDataGas(util::randomUInt256));
+                    .excessDataGas(util::randomUInt64));
```
Comment (gfukushima) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/DataStructureUtilAdapter.java:138]: Is this change related to the teku upgrade?

**Review by gfukushima — APPROVED** [review]

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/DataStructureUtilAdapter.java (hunk 1 above)
Comment (jframe) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/DataStructureUtilAdapter.java:138]: It is, the excessDataGas field is now uint64 instead of unint256. We only have this for some data in tests so far.
